src/psyclone/tests/domain/lfric/lfric_types_test.py

```diff
@@ -40,74 +40,77 @@
 
 import pytest
 
-from psyclone.domain.lfric import LFRicTypes
+from psyclone.domain.lfric import LFRicConstants, LFRicTypes
+from psyclone.errors import InternalError
 from psyclone.psyir.symbols import ContainerSymbol, DataSymbol, \
     ImportInterface, ScalarType, LocalInterface, ArgumentInterface, \
     ArrayType, Symbol
 from psyclone.psyir.nodes import Reference, Literal
 
 
+# pylint complains about all isinstance tests involving LFRicTypes,
+# since it doesn't know what the actual return type is. So disable
+# it here globally for this file:
+# pylint: disable=isinstance-second-argument-not-valid-type
+
+
 def test_singleton(monkeypatch):
     '''Tests that the singleton implementation works as expected.
 
     '''
     # First check that on first time creation the class is properly
     # initialised:
-    monkeypatch.setattr(LFRicTypes, "_instance", None)
-    LFRicTypes()
-    assert LFRicTypes._instance is not None
+    monkeypatch.setattr(LFRicTypes, "_name_to_class", None)
+    assert LFRicTypes._name_to_class is None
+    LFRicTypes("I_DEF")
+    assert LFRicTypes._name_to_class is not None
 
     # Now replace the internal dictionary
-    monkeypatch.setattr(LFRicTypes._instance, "_name_to_class",
+    monkeypatch.setattr(LFRicTypes, "_name_to_class",
                         {"new": "really_new"})
     # Make sure we get the new dictionary:
-    assert LFRicTypes()("new") == "really_new"
+    assert LFRicTypes("new") == "really_new"
 
-    # But if we delete the instance, the singleton will create a
-    # new instance, which does not have 'new' as key anynmore:
-    monkeypatch.setattr(LFRicTypes, "_instance", None)
-    with pytest.raises(KeyError) as err:
-        LFRicTypes()("new")
-    assert "new" in str(err.value)
+    # Check error handling:
+    monkeypatch.setattr(LFRicTypes, "_name_to_class", None)
+    with pytest.raises(InternalError) as err:
+        LFRicTypes("does_not_exist")
+    assert "Unknown type 'does_not_exist'. Valid values are " in str(err.value)
```

Comment:
Perhaps "Unknown LFRic type" in the message just to be a bit clearer.

Reply:
Done.